iroh-bytes/src/store/fs/migrate_redb_v1_v2.rs

```diff
@@ -0,0 +1,321 @@
+use std::path::{Path, PathBuf};
+
+use anyhow::Result;
+use redb_v1::ReadableTable;
+use tempfile::NamedTempFile;
+use tracing::info;
+
+pub fn run(source: impl AsRef<Path>) -> Result<redb::Database> {
+    let source = source.as_ref();
+    // create the database to a tempfile
```

Comment:
I think we should use a tempfile next to the source directory, thinking about multiple disk setups, and the db being large, you want to make sure you stay on the same disk